test/extensions/filters/network/common/multiplexing/BUILD

```diff
@@ -0,0 +1,58 @@
+licenses(["notice"])  # Apache 2
+
+load(
+    "//bazel:envoy_build_system.bzl",
+    "envoy_cc_mock",
+    "envoy_package",
+)
+load(
+    "//test/extensions:extensions_build_system.bzl",
+    "envoy_extension_cc_test",
+    "envoy_extension_cc_test_binary",
+)
+
+envoy_package()
+
+envoy_cc_mock(
+    name = "multiplexing_mocks",
+    srcs = ["mocks.cc"],
+    hdrs = ["mocks.h"],
+    deps = [
+        "//source/common/common:assert_lib",
+        "//source/extensions/filters/network/common/multiplexing:conn_pool_interface",
+        "//source/extensions/filters/network/common/multiplexing:router_interface",
+        "//source/extensions/filters/network/common/redis:client_interface",
```

Comment:
It seems wrong that we are still pointing into Redis in here? Can we tease that apart?

Reply:
I was thinking of removing that dependency in a commit similar to this one https://github.com/envoyproxy/envoy/commit/3dc28c09a4b20a4ff95f2e3fce6670d7978a4b58. I can merge it into this branch and fix the tests if it's preferred to do it all in one go. WDYT?

Reply:
I would discuss with the redis folks in Slack and see what the consensus is. I'm fine with whatever you all want to do, but we should try to keep things clean which would mean no circular dependencies like this.